public/electron/fuse/index.js

```diff
@@ -0,0 +1,101 @@
+const path = require('path');
+const chalk = require('chalk');
+const get = require('lodash/get');
+const { spawn, exec } = require('child_process');
+
+class FuseInstallerProcess {
+  constructor() {
+    this.childProcess = null;
+    this.handlers = {
+      ready: [],
+      failed: [],
+      pending: [],
+    };
+  }
+
+  on(key, handler) {
+    const accHandlers = get(this.handlers, key, []) || [];
+
+    this.handlers = {
+      ...this.handlers,
+      [key]: [
+        ...accHandlers,
+        handler,
+      ],
+    };
+  }
+
+  callHandlers(key, args) {
+    if (this.handlers[key]) {
+      this.handlers[key].forEach((handler) => handler(args));
+    }
+  }
+
+  start(isDev) {
+    if (this.childProcess) return;
+
+    if (process.platform !== 'darwin') {
+      return;
+    }
+
+    let installerPath = path.join(process.resourcesPath, 'FuseInstaller.pkg');
+    if (isDev) {
+      installerPath = path.join(__dirname, '../../../resources/FuseInstaller.pkg');
+    }
+
+    this.childProcess = spawn('sudo', ['-A', 'installer', '-pkg', installerPath, '-target', '/'], {
+      env: {
+        PATH: process.env.PATH,
+        SUDO_ASKPASS: path.join(__dirname, 'askpass.osascript.js'),
+      },
+    });
+    this.childProcess.stdout.on('data', (data) => {
+      const outputLog = data.toString().toLowerCase();
+
+      // eslint-disable-next-line no-console
+      console.log(chalk.blue(outputLog));
+
+      if (outputLog.includes('the install was successful') || outputLog.includes('the upgrade was successful')) {
+        // eslint-disable-next-line no-console
+        console.log(chalk.blue('OsxFuse Kernel Installed'));
+        this.loadFuseKernel();
+      }
+    });
+
+    this.childProcess.stderr.on('data', (data) => {
+      // eslint-disable-next-line no-console
+      console.error(chalk.red(data));
+      this.callHandlers('error');
+    });
+
+    this.callHandlers('pending');
+  }
+
+  loadFuseKernel() {
```

Comment:
do we have to load manually the fuse always? like the daemon?

Reply:
This `loadFuseKernel()` function is a private function that is called inside in the `start()` function. The start function ensures that the kernel is installed first before loading it.

Also, the installer.start() should only be called when a user wants to mount the fuse drive and the status of the fuse drive gotten from the daemon is `NOT_INSTALLED`.